src/gui/guiPasswordChange.h

```diff
@@ -44,6 +44,13 @@ class GUIPasswordChange : public GUIModalMenu
 	bool processInput();
 
 	bool OnEvent(const SEvent &event);
+#ifdef __ANDROID__
+	bool getAndroidUIInput();
+#endif
+
+protected:
+	std::wstring getLabelByID(s32 id) { return L""; }
+	std::string getNameByID(s32 id);
```

Comment:
this function should return const std::string & instead

Reply:
Maybe, but that code was taken from the existing implementation in `guiFormSpecMenu.h` and simply moved to the base class. 

https://github.com/minetest/minetest/blob/79e393cf4b5027eb9071df23f4834aac3f3a689f/src/gui/guiFormSpecMenu.h#L538-L539

If after 3 months you are going to come back with something like this then I feel I am wasting my time. The android build is currently borked anyway, feel free to adopt this PR if you wish to make more style changes.

Reply:
@paramat, he is probably right and I would not have minded if it had been pointed out before now. There are many places in MT code-base where constness is ignored and it is a good idea to fix these issues with new code. I will make these changes when I have time, please forgive me for feeling a bit dejected (if that's the right word)

Reply:
Hey no problem, it's very understandable, contributing when dev and reviews are slow is difficult and frustrating.

Reply:
Yeah, frustrated, that was the word I was looking for :)

I would be willing to do a lot more if I thought it might be processed in weeks rather than months.

Reply:
> this function should return const std::string & instead

@nerzhul I started to do this then realized that we cannot return a const reference to the string literals `L""` & `""` as these are temporary objects.